### ISSUE TITLE
Project fails to run on Python 3.7.2

### ISSUE THREAD

**Issue by harenbrs** [ederag/geoptics#3]
Following the usage instructions in the project readme immediately raises an exception:

```
$ python t_geo.py 
Traceback (most recent call last):
  File "t_geo.py", line 29, in <module>
    gui = gui.Gui()
  File ".../GeOptics/geoptics/guis/qt/main.py", line 83, in __init__
    self.setup()
  File ".../GeOptics/geoptics/guis/qt/main.py", line 193, in setup
    self.read_settings()
  File ".../GeOptics/geoptics/guis/qt/main.py", line 392, in read_settings
    self.restoreState(settings.value("MainwindowState"))
TypeError: restoreState(self, Union[QByteArray, bytes, bytearray], version: int = 0): argument 1 has unexpected type 'NoneType'
```

This is a fresh Python 3.7.2 virtual environment on MacOS 10.13.6 with the following packages:
```
$ pip freeze
PyQt5==5.12.1
PyQt5-sip==4.19.15
PyYAML==5.1
sip==4.19.8
```

Am I doing something wrong? @ederag, could you share the version dependencies of your working environment?

**Comment by ederag** [comment]
Thanks for the clear report.
Here is my environment: 
```
Python 3.6.5
PyYAML==3.12
python3-qt5: 5.10.1-lp150.3.1
python3-sip: 4.19.7-lp150.1.3
```
i'll try to set up a python-3.7 tox environment tomorrow.

**Comment by ederag** [comment]
```bash
tox -e py37  # tests run fine
. .tox/py37/bin/activate
python3.7 t_geo.py  # main window appears, similar to the screenshot
pip3.7 list
Package        Version
-------------- -------
atomicwrites   1.3.0  
attrs          19.1.0 
GeOptics       0.19   
more-itertools 7.0.0  
pip            19.0.3 
pluggy         0.9.0  
py             1.8.0  
PyQt5          5.12.1 
PyQt5-sip      4.19.15
pytest         4.4.0  
PyYAML         5.1    
setuptools     41.0.0 
six            1.12.0 
wheel          0.33.1 
```
So it does not seem related to python3.7, 
but maybe rather to the fact that there is no configuration to restore for a new user ?
That would explain the `NoneType`.
Indeed, I can reproduce it with a new user.

**Comment by ederag** [comment]
Should be solved by PR #4 (`restore_state` branch).

**Comment by harenbrs** [comment]
Indeed it is! Many thanks for the quick fix.
